datapackage_pipelines_assembler/processors/load_preview.py

```diff
@@ -4,6 +4,8 @@
 
 
 def generate_preview(res):
+    if len(list(res)) < int(parameters.get('limit', 10000)):
```

Comment:
This won't work - you're consuming the resource here, so you can't iterate over it later on.
Also, `list(res)` will load the entire thing into memory - which is not a good idea.

You should rely on the `rowCount` of the resource and skip it in process_resources.

Reply:
In `process_resources`, check `res.spec['rowCount']` and decide whether to `yield generate_preview(res)` or `yield []`.

Reply:
This makes sense, just getting `Output pipe disappeared!` error when yielding empty list

Reply:
Eg this error:
```
if rowcount > limit:
            yield generate_preview(res)
        else:
            yield []
```

But this works fine:

```
if rowcount > limit:
            yield generate_preview(res)
        else:
            a = list(res)
            a = None
            yield []
```

Reply:
yes, you are correct - you need to consume the source (even if you're not using it).

The best way to do this is
`collections.deque(res, maxlen=0)`
which doesn't take up memory space